crypto/bls12381_utils.c

```diff
@@ -580,33 +570,38 @@ void ep_sum_vector(ep_t jointx, ep_st* x, const int len) {
 }
 
 // Computes the sum of the signatures (G1 elements) flattened in a single sigs array
-// and store the sum bytes in dest.
+// and writes the sum (G1 element) as bytes in dest.
 // The function assumes sigs is correctly allocated with regards to len.
 int ep_sum_vector_byte(byte* dest, const byte* sigs_bytes, const int len) {
+    int error;
+
     // temp variables
     ep_t acc;        
     ep_new(acc);
     ep_set_infty(acc);
-    ep_st* sigs = (ep_st*) malloc(len * sizeof(ep_st));
+    ep_st* sigs = (ep_st*) calloc(len, sizeof(ep_st));
```

Comment:
Not sure why are we using `calloc`. I don't see that we rely on 0 initialized memory, in this case malloc will be faster since calloc does a `memset(0)` which is usually slower.

Reply:
The main reason for using `calloc` here is to protect against `len * sizeof(ep_st)` overflow.  If we consider that it won't practically happen in our setup (64 bit, limited `len`) then I can likely leave `malloc` here.

(there is a separate issue for verifying the `malloc` returns against `NULL` in the codebase)

Reply:
I actually missed this line change during my review. 
The results of my napkin-math says that if there is overflow, the input `sigs_bytes` should point to a block of data of 3 EiB (Exbibytes), which is more than 3,000,000 Terabytes. 

I would rather avoid `memset` for every small data, than support the feature for such unrealistic size. I think `malloc` seems fine in this case.

Reply:
re: `calloc` it was designed to help with security of 32bit platforms since it does fail on multiplication overflow, `malloc(3)`:

> If the multiplication of nmemb and size would result in integer
> overflow, then calloc() returns an error.  By contrast, an
> integer overflow would not be detected in the following call to
> malloc(), with the result that an incorrectly sized block of
> memory would be allocated:
> malloc(nmemb * size);

As for the allocation on stack, i'm a bit scared about it, unless the upper bound is in the low kilobytes: even with a pretty large stack there is no way to guarantee that this function won't be called from a recursive function, which there are some in our codebase.

Reply:
Overflow in `malloc` happens if `len * sizeof(ep_st)` exceeds 64 bits while `len` is less than 64 bits. `calloc` takes `len` as a separate input without the need to multiply.

We do no have an upper bound for `len` though we could add one for safety. At the same time, values of `len` creating issues do not seem practical.